feat: Sort the models of each provider by release_date (from the most recent to the oldest)

AAEE86 opened the pull request.

AAEE86 commented:
@looplj 其实我觉得一些过时的模型可以删除

looplj commented:
需要修改：
https://github.com/looplj/axonhub/blob/unstable/scripts/sync-model-developers.js 

这个脚本，不然会被覆盖，排除逻辑也可以放到这里

AAEE86 committed:
feat(scripts): add model sorting by release date

Add sortModelsByDate function to sort models within each provider by release date in descending order (newest first). Models without release dates are treated as oldest.

AAEE86 commented:
> 需要修改： https://github.com/looplj/axonhub/blob/unstable/scripts/sync-model-developers.js
> 
> 这个脚本，不然会被覆盖，排除逻辑也可以放到这里

改了脚本你看下

looplj commented:
LGTM
